docs/core/deploying/trimming/prepare-libraries-for-trimming.md

```diff
@@ -8,360 +8,276 @@ ms.date: 06/12/2023
 
-  <PropertyGroup>
-    <OutputType>Exe</OutputType>
-    <TargetFramework>net7.0</TargetFramework>
-    <PublishTrimmed>true</PublishTrimmed>
-  </PropertyGroup>
+### .csproj file
 
-  <ItemGroup>
-    <ProjectReference Include="path/to/MyLibrary.csproj" />
-    <!-- Analyze the whole library -->
-    <TrimmerRootAssembly Include="MyLibrary" />
-  </ItemGroup>
+:::code language="xml" source="~/docs/core/deploying/trimming/snippets/ConsoleApp1/ConsoleApp1.csproj":::
 
-</Project>
-```
+**Notes:**
+
+* In the preceding project file, when using .NET 7, replace `<TargetFramework>net8.0</TargetFramework>` with `<TargetFramework>net7.0</TargetFramework>`.
+* [`<TrimMode>full</TrimMode>`](/dotnet/core/compatibility/deployment/7.0/trim-all-assemblies#recommended-action) in the `<PropertyGroup>` tag:
+
+  * Is the [default for .NET 7 and higher](../../../core/compatibility/deployment/7.0/trim-all-assemblies.md).
+  * Is not shown in the preceding project file.
+  * Ensures that the trimmer only analyzes the parts of the library's dependencies that are used.
+  * Tells the trimmer that any code that isn't part of a "root" can be trimmed if it's unused. Without this   option:
+    * Warnings are issued originating from ***any*** part of a dependency that doesn't set `[AssemblyMetadata  ("IsTrimmable", "Tue")]`
+    * The preceding warnings can be issued for code that is unused by the library.
 
 ### [.NET 8+](#tab/net8plus)
 
-```xml
-<Project Sdk="Microsoft.NET.Sdk">
+<!-- exact duplicate of tab7 but forced by builder to have identical tabs -->
+* Add `<PublishTrimmed>true</PublishTrimmed>`.
+* Add a reference to the library project with `<ProjectReference Include="/Path/To/YourLibrary.csproj" />`.
+* Specify the library as a trimmer root assembly with `<TrimmerRootAssembly Include="YourLibraryName" />`.
+  * `TrimmerRootAssembly` ensures that every part of the library is analyzed. It tells the trimmer that this assembly is a "root". A "root" assembly means the trimmer analyzes every call in the library and traverses all code paths that originate from that assembly.
 
-  <PropertyGroup>
-    <OutputType>Exe</OutputType>
-    <TargetFramework>net8.0</TargetFramework>
-    <PublishTrimmed>true</PublishTrimmed>
-  </PropertyGroup>
+### .csproj file
 
-  <ItemGroup>
-    <ProjectReference Include="path/to/MyLibrary.csproj" />
-    <!-- Analyze the whole library -->
-    <TrimmerRootAssembly Include="MyLibrary" />
-  </ItemGroup>
+:::code language="xml" source="~/docs/core/deploying/trimming/snippets/ConsoleApp1/ConsoleApp1.csproj":::
 
-</Project>
-```
+**Notes:**
+
+* [`<TrimMode>full</TrimMode>`](/dotnet/core/compatibility/deployment/7.0/trim-all-assemblies#recommended-action) in the `<PropertyGroup>` tag:
```

Comment:
Personally I find this confusing - the project file doesn't have that property (and we do note that here), but initially I was looking for it (since I didn't get to the note that it's not there yet). After reading it all I was like "Why do I care?" - it's the default, I want all of this, so why are you telling me this.

The final bullet point about the "Tells the trimmer" is really confusing, because it's a negation - the subpoints are NOT true in the test sample app, but to know that one has to fully read and understand the sentence before them... I honestly had to read this like 3 times to understand what's going on.

I think this description is a good one to have to document the `TrimMode` property itself, but I would not include it in the description of the test app.